ncm-opennebula/src/main/perl/opennebula.pm

```diff
@@ -1,42 +1,361 @@
+Modify your machine template to:
+
+=over 4
+
+=item * Rename hdx/sdx device disks by vdx to use virtio module
+
+=item * Create or include a new pan file to:
+
+=over 5
+
+=item * Include 'metaconfig/opennebula/schema'
+
+=item * Set the OpenNebula opennebula/datastore name for each vdx
+
+=item * Set the VNETs opennebula/vnet (bridges) required by each VM network interface
+
+=item * (Optional) Set /system/opennebula/ignoremac tree to avoid to include MAC values within AR/VM templates
+
+=item * (Optional) Set /system/opennebula/graphics to export VM graphical display (VNC is used by default)
+
+=item * (Optional) Set /system/opennebula/labels list to include OpenNebula search labels
+
+=item * (Optional) Set /system/opennebula/pci list values to enable PCI Passthrough
+
+=over 6
+
+=item * PCI Passthrough section is also generated based on hardware/cards/<card_type>/<interface>/pci values
+
+=back
+
+=item * (Optional) Set /system/opennebula/diskcache to select the cache mechanism for your disks. (by default is set to none)
+
+=item * (Optional) Set OPENNEBULA_AII_REPLACE_MAC and MAC_PREFIX variables to replace VM MACs to use OpenNebula style
+
+=back
+
+=item * enable acpid service
+
+=back
+
+The next paths are relative to /system/aii/hooks/configure
+
+=over 4
+
+=item * image : boolean
+
+Create VM images. Note: VM images are not updated, if you want to resize or modify an available image from scratch use remove hook first.
+
+=item * template : boolean
+
+Create VM template.
+
+=back
+
+=head4 install
+
+Instantiates the required VM. 
+The next paths are relative to /system/aii/hooks/install
+
+=over 4
+
+=item * vm : boolean
+
+The new VM is instantiated after resources creation
+
+=item * onhold : boolean
+
+The VM is instantiated but onhold status instead of running
+
+=back
+
+=head4 remove
+
+Removes the VM resources (template/images/network). 
+All these paths are relative to /system/aii/hooks/remove
+
+=over 4
+
+=item * vm : boolean
+
+Stop running VM
+
+=item * image : boolean
+
+Remove image/s
+
+=item * template : boolean
+
+Remove VM templates
+
+=back
+
+=head1 DEPENDENCIES
+
+The component was tested with OpenNebula version 4.1x and 5.x
+
+Following package dependencies should be installed to run the component:
+
+=over
+
+=item * aii-ks
+
+=item * perl-Config-Tiny
+
+=item * perl-LC
+
+=item * perl-CAF
+
+=item * perl-Set-Scalar
+
+=item * perl-Net-OpenNebula >= 0.2.2 !
+
+=back
+
+=head1 METHODS
+
+=cut
+
+use parent qw(NCM::Component
+              NCM::Component::OpenNebula::Commands
+              NCM::Component::OpenNebula::Host
+              NCM::Component::OpenNebula::Ceph
+              NCM::Component::OpenNebula::Server
+              NCM::Component::OpenNebula::Account
+              NCM::Component::OpenNebula::Network
+              NCM::Component::OpenNebula::VM
+              NCM::Component::OpenNebula::Image
+              );
+use NCM::Component::OpenNebula::Commands;
```

Comment:
these are not needed (excpet for the import of constants). this is waht the `use parent` does